src/presets.rs

```diff
@@ -380,6 +381,12 @@ mod tests {
             }
         }
     }
+
+    #[test]
+    fn generate_prime_into_int_double_size_precision() {
```

Comment:
I am not sure if we should add regression tests for bugs in third-party libraries (perhaps it can temporary and be removed when crypto-bigint publishes a version where it's fixed?). But if you think we do:
- This needs some comment about what's actually being tested
- The test may be simplified to 
   ```
   let num = U256::from_be_hex("00000000000000000000000000000000E0BD7D7A037746B47ABDDC7065251485");
   assert!(is_prime(&num));
   // or even
   assert_eq!(lucas_test(&Odd::new(num).unwrap(), AStarBase, LucasCheck::Strong), Primality::ProbablyPrime);
   ```

Reply:
This bug was not caught by our test suite and only showed in one particular benchmark; we do not run benchmarks in CI and they take some time to run so it's easy for problems that show up in benches to go unnoticed for a long time. When prime generation hangs, the consequences upstream could be bad (I think the whole app will hang unless it's coded defensively using async tasks with deadlines).

So that was the main motivation here: a test that exercise the code when asking for an n-bit prime in a 2n-bit wide uint.

That said, I agree with you that this test is poorly motivated and written. Will improve it!